File: Leetcode-C++/71_SimplifyPath.cpp
class Solution {
public:
    string simplifyPath(string path) {
        stack<string> st;
        string res;
        string temp;
        
        for(int i = 0;  i<path.size(); ++i)
        {
            if(path[i] == '/')    
                continue;
            temp.clear();
            while(i < path.size() && path[i] != '/')
            {
                temp += path[i++];
            }
            if(temp == ".")
                continue;
            else if(temp == "..")
            {
                if(!st.empty())
                    st.pop();
            }
            else
                st.push(temp);
        }
        
        while(!st.empty())
        {
            res = "/" + st.top() + res;
            st.pop();
        }
        
        return res.size()==0 ? "/" : res;
    }
};

File: Leetcode-Python/71_SimplifyPath.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        splitPath = [i for i in path.split('/') if i !='']
        ans = []
        for p in splitPath:
            if p.isalpha():
                ans.append(p)
            elif p =='..':
                if ans != []:
                    ans.pop()
            elif p =='.':
                pass
            else:
                ans.append(p)
        return '/'+'/'.join(ans)



# class Solution:
#     def simplifyPath(self, path: str) -> str:
#         ans = []
#         word = ''
#         idx = 0
#         for i in range(len(path)-1,0,-1):
#             if path[i].isalpha():
#                 idx = i
#                 break
        
#         path = path[:idx+1]+'/'
  
#         prev = path[0]
#         for p in path[1:]:
#             if p.isalpha():
#                 word += p
#             if p =='.' and len(ans)!=0:
#                 ans =[]
#             if prev.isalpha() and p =='/':
#                 ans.append(p)
#                 ans.append(word)
#                 word = ''
#             prev = p
#         return ''.join(ans) if ans else '/'